src/views/webview-app/use-connection-form.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import type { ConnectionInfo } from 'mongodb-data-service-legacy';
+import { sendConnectToExtension } from './vscode-api';
+import { MESSAGE_TYPES } from './extension-app-message-constants';
+import type { MESSAGE_FROM_EXTENSION_TO_WEBVIEW } from './extension-app-message-constants';
+
+export default function useConnectionForm() {
```

Comment:
I think we want to track if the connection attempt is in progress and disable/loader the `Connect` button in that instance. Right now if you open the form, turn off internet, and then connect to a cluster on the web there's no visual feedback.